x-pack/platform/plugins/shared/lens/public/datasources/text_based/text_based_languages.tsx

```diff
@@ -235,8 +235,10 @@ export function getTextBasedDatasource({
         );
         return {
           columnId: c.variable ?? c.id,
-          fieldName: c.variable ? `?${c.variable}` : c.name,
+          fieldName: c.variable ? `?${c.variable}` : c.id,
           variable: c.variable,
+          label: c.name,
+          customLabel: c.id !== c.name,
```

Comment:
Can it be `hasCustomLabel`?

Noticed that the custom label gets reset when interval changes. Not sure if it's okay.

Reply:
This is an already existing property so I cant rename it unfortunately. 

The custom label will reset here yes but I think that is ok as it happens only for the discover histogram. I dont expect users to want to give a custom label tbh. If we think that they need to, then we need to find another solution and not rely on the custom label of the Lens charts.